backend/db/repository.go

```diff
@@ -23,6 +24,17 @@ func NewUserRepository(db *sql.DB) UserRepository {
 }
 
 func (r *UserDBRepository) AddUser(ctx context.Context, user domain.User) (int64, error) {
+	// Check if user already exists
+	var exists bool
```

Comment:
I might be wrong, but we may not need this because our `user` table, `name` column has `UNIQUE` constraint. So if the user already exists, a duplicate `INSERT` query will return an error

Reply:
OK, just committed the new one